pkg/specgen/generate/container_create.go

```diff
@@ -486,5 +475,36 @@ func createContainerOptions(ctx context.Context, rt *libpod.Runtime, s *specgen.
 	if s.PidFile != "" {
 		options = append(options, libpod.WithPidFile(s.PidFile))
 	}
+
 	return options, nil
 }
+
+func Inherit(infra libpod.Container) (opts []libpod.CtrCreateOption, infraS *spec.Spec, compat *libpod.InfraInherit, err error) {
+	options := []libpod.CtrCreateOption{}
+	compatibleOptions := &libpod.InfraInherit{}
+	infraConf := infra.Config()
+	infraSpec := infraConf.Spec
+
+	config, err := json.Marshal(infraConf)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	err = json.Unmarshal(config, compatibleOptions)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if infraSpec.Linux != nil && infraSpec.Linux.Resources != nil {
+		resources, err := json.Marshal(infraSpec.Linux.Resources)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		err = json.Unmarshal(resources, &compatibleOptions.InfraResources)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+	}
+	if compatibleOptions != nil {
+		options = append(options, libpod.WithInfraConfig(*compatibleOptions))
```

Comment:
Instead of doing this, can we return the InfraInherit and do the unmarshal over the spec within pkg/specgen/generate/oci.go as we create the spec itself? Localizes the changes to pkg/spec which seems simpler, though you will have to pass the InfraInherit around a bit

Reply:
Actually, it looks like you're passing it into SpecGenToOCI below, so it probably isn't even necessary to change how it's passed around

Reply:
so is this good as is? the`SpecGenToOCI` bit is just for config elements that get overwritten. This works for like 90% of options.

Reply:
I'd prefer to remove the option and just set the values in the spec in SpecGenToOCI - most options alter more than just the spec, if we're just changing the spec we can do it during spec generation

Reply:
ok, I am trying to have this just be done in the spec, but for some reason all of the tests are failing even though the runtime spec has the proper values. I am going to investigate a bit,

Reply:
Huh...

Can you verify if it has the correct values after it gets put into the DB? I wonder if some other part of container creation is overwriting it.

Reply:
ok @mheon I did some digging, I do not think the issue is with the runtime spec, I think the container config needs to be modified as well. some options like volumes are not explicitly in the spec and will be dropped. 

Instead of doing options, I could make a path that opts to do something like a checkpoint where we load the config from infra (or at least what matches). Could help delineate this from regular create options

Reply:
ok @mheon, then is this PR good as is? I never pushed the new changes.